cdk/visit/lambda_code/register_user/register_user.py

```diff
@@ -34,7 +34,7 @@ def addUserInfo(user_info):
     return response['ResponseMetadata']['HTTPStatusCode']
 
 
-def handler(request, context):
+def handler(request, context, table=users):
```

Comment:
Have you considered an object oriented approach to this? e.g.

```
class RegisterUserFunction():
    def __init__(self, users):
        if table is None:
            # Get the service resource.
            dynamodb = boto3.resource('dynamodb')
            # Get the table name.
            TABLE_NAME = os.environ["TABLE_NAME"]
            # Get table objects
            self.users = dynamodb.Table(TABLE_NAME)
        else: 
            self.users = users

    def handle_register_user_request(self, request, context):
        # implement business logic here

fn = RegisterUserFunction(users)

# Entry point of our lambda function remains the same, but we now have a handler object that 
# encapsulates the logic necessary to register the user
handler(request, context):
  fn.handle_register_user_request(request, context)
```

Through these means you should be able to leave the rest of the function interfaces as-is and your tests can be written against the RegisterUserFunction class methods.

Reply:
I did not originally. At the time, I wanted to go with an approach that minimized the changes to the handler function and the approach I took was the first one that came to mind. I definitely do see the merit in the approach you describe though. @ophilli What are your thoughts? I'm 50/50 on which approach I prefer.

Reply:
Just implemented it in the latest commit!